presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -154,6 +155,7 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
                                         new PushLimitThroughSemiJoin(),
                                         new MergeLimitWithDistinct(),
 
+                                        new PruneSemiJoinColumns(),
```

Comment:
The output pruning rules have been there since Martin started adding them here: 69b3c7f
I was just adding more in the same place.